### ISSUE TITLE
[BUG] F-Droid reproducible build failed

### ISSUE THREAD

**Issue by linsui** [vitorpamplona/amethyst#1113]
1. The tor-android libs are from github which is not allowed for F-Droid. I tried to build it from source but the lib is not reproducible.
2. Besides that, the apk is not reproducible. https://gitlab.com/linsui/fdroiddata/-/jobs/7946708164 Is the apk built from the tag?

**Comment by jivanpal** [comment]
@vitorpamplona Please prioritise this if reasonable to do so. F-Droid users are currently stuck on v0.91.0 due to build failures.

**Comment by vitorpamplona** [comment]
I took a look some days ago, but I have no idea what's creating the lack of reproducibility. 

> The tor-android libs are from github which is not allowed for F-Droid

Maybe we need to give up on F-droid since I don't think there is another way to use tor without the library.

**Comment by licaon-kter** [comment]
Is this because @n8fr8 ain't pushing to Maven Central too? https://mvnrepository.com/artifact/info.guardianproject/tor-android

**Comment by linsui** [comment]
Yes, as always.

> Maybe we need to give up on F-droid since I don't think there is another way to use tor without the library.

You can build it from source.

**Comment by licaon-kter** [comment]
So the app is reproducible still...

**Comment by eighthave** [comment]
I think the path of least resistance would be to update tor-android on Maven Central, if we can get the maintainers to do so.

FYI @uniqx @n8fr8

**Comment by licaon-kter** [comment]
we've disabled autoupdates checks for now, https://gitlab.com/fdroid/fdroiddata/-/commit/c010648f2963a5393d9d9f0e994421bd7d55ba84

hopefully this will get fixed somehow soon

**Comment by vitorpamplona** [comment]
@licaon-kter we have moved away from that Tor library and using Arti Tor that we compile ourselves. I also checked and it looks like we are not fully reproducible. 

Can you check if we are now compliant with your policies?

**Comment by jivanpal** [comment]
@vitorpamplona said:

> looks like we are not fully reproducible

Typo? Intended to say "*now* fully reproducible"?

**Comment by vitorpamplona** [comment]
Oops... Yes we seem to be fully reproducible now

**Comment by linsui** [comment]
Then https://github.com/vitorpamplona/amethyst/blob/main/settings.gradle.kts#L28 can be removed?

**Comment by vitorpamplona** [comment]
Removed it: https://github.com/vitorpamplona/amethyst/pull/4118

**Comment by linsui** [comment]
Which ndk is used? https://github.com/vitorpamplona/amethyst/tree/main/tools/arti-build mentions r27 but a r25 version is used instead.

**Comment by linsui** [comment]
And I thought there is another ndk version used to strip the lib. Please pin the ndk version and make sure it's the only version installed and used.

**Comment by vitorpamplona** [comment]
Pinned: https://github.com/vitorpamplona/amethyst/pull/4120

**Comment by linsui** [comment]
I guess there is no apk we can check yet?

**Comment by linsui** [comment]
> And I thought there is another ndk version used to strip the lib. Please pin the ndk version and make sure it's the only version installed and used.

This is not fixed. There is still another ndk version used to strip the lib.

**Comment by linsui** [comment]
Yes, comfirmed that r28c is used to strip the libs. Please use the same ndk version everywhere.

**Comment by linsui** [comment]
And I'm not sure why there are also armeabi-v7a and x86 libs in the apk though arti is not built for them.

**Comment by vitorpamplona** [comment]
Cool, 
APK issue: https://github.com/vitorpamplona/amethyst/pull/4144
NDL: https://github.com/vitorpamplona/amethyst/pull/4142

**Comment by linsui** [comment]
I'll update the metadata again when the next version is published. 1.16.0 will be published soon. Thanks!

**Comment by vitorpamplona** [comment]
Cool! I will do one soon just to test
